sdk/main/include/ECDSAsecp256k1PrivateKey.h

```diff
@@ -147,28 +144,29 @@ class ECDSAsecp256k1PrivateKey : public PrivateKey
 
 private:
   /**
-   * Create an OpenSSL keypair object from a byte vector representing an ECDSAsecp256k1PrivateKey.
+   * Create an OpenSSL key object from a byte vector representing an ECDSAsecp256k1PrivateKey.
    *
    * @param keyBytes The bytes representing a ECDSAsecp256k1PrivateKey.
    * @return A pointer to a newly created OpenSSL keypair object.
    */
-  static EVP_PKEY* bytesToPKEY(const std::vector<unsigned char>& keyBytes);
+  static std::unique_ptr<EVP_PKEY, void (*)(EVP_PKEY*)> bytesToPKEY(const std::vector<unsigned char>& keyBytes);
 
```

Comment:
Have you considered making a wrapper class around EVP_PKEY (and any other openssl object you follow this custom deleter pattern for)? The custom deleter logic could be relegated to the wrapper classes and thoroughly explained in just 1 spot

Reply:
Is `std::unique_ptr` not the wrapper class? I don't know if I fully understand what you're suggesting

Reply:
That was the thought - a c++ class containing only the unique pointer, which knows how to safely destruct. May or may not be a good idea

Reply:
Pushed up an implementation, let me know what you think